WordPress/src/androidTest/java/org/wordpress/android/ui/screenshots/WPLocaleTestRule.java

```diff
@@ -1,57 +1,53 @@
 package org.wordpress.android.ui.screenshots;
 
+import android.content.Context;
 
 import androidx.test.core.app.ApplicationProvider;
-import androidx.test.platform.app.InstrumentationRegistry;
 
 import org.junit.rules.TestRule;
 import org.junit.runner.Description;
 import org.junit.runners.model.Statement;
 import org.wordpress.android.WordPress;
 import org.wordpress.android.util.LocaleManager;
 
-public class WPLocaleTestRule implements TestRule {
-    private static final String FASTLANE_TEST_LOCALE_KEY = "testLocale";
-    private static final String FASTLANE_ENDING_LOCALE_KEY = "endingLocale";
+import tools.fastlane.screengrab.locale.LocaleUtil;
 
+public class WPLocaleTestRule implements TestRule {
     private String mTestLocaleCode;
-    private String mEndLocaleCode;
 
     public WPLocaleTestRule() {
-        this(localeCodeFromInstrumentation(FASTLANE_TEST_LOCALE_KEY),
-                localeCodeFromInstrumentation(FASTLANE_ENDING_LOCALE_KEY));
+        this(LocaleUtil.getTestLocale());
     }
 
-    public WPLocaleTestRule(String testLocaleCode, String endLocaleCode) {
+    public WPLocaleTestRule(String testLocaleCode) {
         mTestLocaleCode = testLocaleCode;
-        mEndLocaleCode = endLocaleCode;
     }
 
     @Override
     public Statement apply(final Statement base, Description description) {
         return new Statement() {
             @Override
             public void evaluate() throws Throwable {
+                String original = null;
                 try {
                     if (mTestLocaleCode != null) {
-                        changeLocale(mTestLocaleCode);
+                        original = changeLocale(mTestLocaleCode);
                     }
                     base.evaluate();
                 } finally {
-                    if (mEndLocaleCode != null) {
-                        changeLocale(mEndLocaleCode);
+                    if (original != null) {
+                        changeLocale(original);
                     }
                 }
             }
         };
     }
 
-    private static void changeLocale(String localeCode) {
-        LocaleManager.setNewLocale(ApplicationProvider.getApplicationContext(), localeCode);
-        WordPress.updateContextLocale();
-    }
-
-    private static String localeCodeFromInstrumentation(String key) {
-        return InstrumentationRegistry.getArguments().getString(key);
+    private static String changeLocale(String localeCode) {
+        Context context = ApplicationProvider.getApplicationContext();
+        String original = LocaleManager.getLanguage(context);
+        LocaleManager.setNewLocale(context, localeCode);
+        WordPress.updateContextLocale(ApplicationProvider.getApplicationContext());
```

Comment:
NP: same `context` captured in the variable can be passed here to DRY the code 👍

Reply:
🤦 Good catch… especially since I initially created the variable exactly to DRY the code in the intention to use it for all three calls… not sure why I missed the last one 😅